Wikipedia/Code/ArticleToolbarController.swift

```diff
@@ -0,0 +1,88 @@
+
+import UIKit
+
+protocol ArticleToolbarHandling: class {
+    func toggleSave(from controller: ArticleToolbarController, shouldSave: Bool)
+    func saveButtonWasLongPressed(from controller: ArticleToolbarController)
+    func showThemePopover(from controller: ArticleToolbarController)
+}
+
+class ArticleToolbarController: Themeable {
+    weak var delegate: ArticleToolbarHandling?
+    
+    let toolbar: UIToolbar
+    private var isSaved: Bool = false //tonitodo: better state handling here
```

Comment:
We could probably clean this out now and have all state flow through setSavedState. This was just a quick way for me to toggle before for testing.